src/Source/DoctrineORMQuerySourceIterator.php

```diff
@@ -49,10 +48,6 @@ public function current()
 
     final public function rewind(): void
     {
-        if ($this->iterator) {
-            throw new InvalidMethodCallException('Cannot rewind a Doctrine\ORM\Query');
-        }
-
         $this->iterator = $this->query->iterate();
```

Comment:
Now I see this, [`Query::iterate` was deprecated in `doctrine/mongo-odm` `1.3`](https://github.com/doctrine/mongodb-odm/blob/1.3.x/lib/Doctrine/ODM/MongoDB/Query/Query.php#L413), we should add a constraint and change the method (in another PR). 

AFAIK, in the ODM returns a [Cursor which is rewindable](https://github.com/doctrine/mongodb/blob/master/lib/Doctrine/MongoDB/Cursor.php#L529).

Reply:
Is this PR ok for you then ?

You made your comment on `DoctrineORMQuerySourceIterator`, but the deprecation is on ODM.

Do you want to create the PR with the migration from iterate to getIterator ?
We also have 0 test about `ODM`, maybe it could be great to add some.